nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -226,6 +242,11 @@ private ObjectWriter getObjectWriter(ObjectMapper mapper, String ppSetting) {
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if (!context.hasIncomingConnection() && (context.getProperty(QUERY) == null)) {
+            throw new RuntimeException("Without an incoming connection, the Query property must be set.");
```

Comment:
The RuntimeException here and on line 276 should be ProcessException, it's a NiFi-specific runtime exception that indicates we realize something went wrong but we don't know how to handle any input/output we might have (like routing to failure).  Also rather than `context.getProperty(QUERY) == null` since there will be a PropertyValue object returned, you should use `context.getProperty(QUERY).isSet()`.

Reply:
Done.

Reply:
I just realized that this block to check for an incoming connection and the Query property will get executed on each onTrigger. I believe it should be done during setup (with an OnScheduled method), check ExecuteSQL for an example. Otherwise this processor can generate an exception on each execution, and if the Run Schedule is zero seconds...